src/Missings.jl

```diff
@@ -208,6 +208,104 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+struct SpreadMissings{F} <: Function
+    f::F
+end
+
+function (f::SpreadMissings{F})(x) where {F}
+    @assert x isa AbstractVector
+
+    if x isa AbstractVector{>:Missing}
+        nonmissinginds = collect(eachindex(skipmissing(x)))
+        res = f.f(view(x, nonmissinginds))
+
+        out = Vector{Union{Missing, eltype(res)}}(missing, length(x))
+        out[nonmissinginds] .= res
+
+        return out
+    else
+        return f.f(x)
+    end
+end
+
+function (f::SpreadMissings{F})(xs...) where {F}
+    if any(x -> x isa AbstractVector{>:Missing}, xs)
+        vecs = Base.filter(x -> x isa AbstractVector, xs)
```

Comment:
```suggestion
        vecs = filter(x -> x isa AbstractVector, xs)
```

Reply:
Seems needed for some reason, so not changed.